lib/beaker/options/parser.rb

```diff
@@ -345,6 +347,29 @@ def normalize_args
 
       end
 
+      # Takes tag arguments given at this point, which are strings, converts
+      # them into the proper format (an array of strings) for Beaker use,
+      # and lastly, validates them.
+      #
+      # @return nil
+      # @api public
+      def normalize_and_validate_tags()
+        @options[:tag_includes] ||= ''
+        @options[:tag_excludes] ||= ''
+        @options[:tag_includes] = @options[:tag_includes].split(',') if @options[:tag_includes].respond_to?(:split)
+        @options[:tag_excludes] = @options[:tag_excludes].split(',') if @options[:tag_excludes].respond_to?(:split)
+        @options[:tag_includes].map!(&:downcase)
+        @options[:tag_excludes].map!(&:downcase)
+
+        #check that a tag is not both included & excluded sets
+        @options[:tag_includes].each do |included_tag|
+          @options[:tag_excludes].each do |excluded_tag|
+            parser_error "tag '#{included_tag}' cannot be in both the included and excluded tag sets" \
+              if included_tag == excluded_tag
```

Comment:
Another approach would be to check for set intersection here:
http://ruby-doc.org/stdlib-2.2.2/libdoc/set/rdoc/Set.html#method-i-intersect-3F